File: core-util/src/main/java/org/signal/core/util/logging/NoopLogger.java
package org.signal.core.util.logging;

/**
 * A logger that does nothing.
 */
class NoopLogger extends Log.Logger {
  @Override
  public void v(String tag, String message, Throwable t, boolean keepLonger) { }

  @Override
  public void d(String tag, String message, Throwable t, boolean keepLonger) { }

  @Override
  public void i(String tag, String message, Throwable t, boolean keepLonger) { }

  @Override
  public void w(String tag, String message, Throwable t, boolean keepLonger) { }

  @Override
  public void e(String tag, String message, Throwable t, boolean keepLonger) { }

  @Override
  public void flush() { }
}

File: core-util-jvm/src/main/java/org/signal/core/util/logging/NoopLogger.kt
/*
 * Copyright 2023 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.signal.core.util.logging

/**
 * A logger that does nothing.
 */
internal class NoopLogger : Log.Logger() {
  override fun v(tag: String, message: String?, t: Throwable?, keepLonger: Boolean) = Unit
  override fun d(tag: String, message: String?, t: Throwable?, keepLonger: Boolean) = Unit
  override fun i(tag: String, message: String?, t: Throwable?, keepLonger: Boolean) = Unit
  override fun w(tag: String, message: String?, t: Throwable?, keepLonger: Boolean) = Unit
  override fun e(tag: String, message: String?, t: Throwable?, keepLonger: Boolean) = Unit
  override fun flush() = Unit
}
